test/routes/group.test.js

```diff
@@ -0,0 +1,86 @@
+import { jest } from "@jest/globals"; // eslint-disable-line import/no-extraneous-dependencies
+import request from "supertest";
+import app from "#app";
+import connector from "#models/databaseUtil";
+import groupModel from "#models/group";
+
+jest.mock("#util");
+
+let server;
+let agent;
+
+beforeAll((done) => {
+  server = app.listen(null, () => {
+    agent = request.agent(server);
+    connector.set("debug", false);
+    done();
+  });
+});
+
+function cleanUp(callback) {
+  groupModel
+    .remove({
+      title: "Group 1",
+      student: "64fdc67feca8a69f01b33614",
+    })
+    .then(() => {
+      connector.disconnect((DBerr) => {
+        if (DBerr) console.log("Database disconnect error: ", DBerr);
+        server.close((serverErr) => {
+          if (serverErr) console.log(serverErr);
+          callback();
+        });
+      });
+    });
+}
+
+afterAll((done) => {
+  cleanUp(done);
+});
+
+describe("group API", () => {
+  it("should create group", async () => {
+    const response = await agent.post("/group/add").send({
+      title: "Group 1",
+      student: "64fdc67feca8a69f01b33614",
+    });
+
```

Comment:
check content-type for post and update operations is it is json or not
eg :- expect(response.headers["content-type"]).toMatch(/json/);
refer accreditation.test.js

Reply:
Whoops! missed that...
Added the  necessary checks now :)
Thanks...